### PR TITLE
   [FIX] website: Error message on website if dynamic snippet category deleted

### PR DESCRIPTION
**Pull request by nboulif**
Issue
  
      - Install "Ecommerce" app
      - Go to 'Shop' page and edit it
      - Add a "Dynamic Products" block
      - Click on it to edit it :
      -  Select any template
      -  Select `Desks` as product category
      -  Save
      - Go to "Website -> Configuration -> eCommerce Categories"
      - Delete the "Desks" category
      - Go back to Shop page
      
      Error : "Record does not exist or has been deleted".

Cause
    
      The search domain is looking for child categories of dataset.productCategoryId
      (who does not exist anymore in the above case).
      `child_of` do a browse() to fetch the records, and therefore generate
      an error if category not found.
        
Solution
      
      Log warning message if unexptected domain and return empty records set instead.
        
opw-2440917

### CODE REVIEW

**Comment by JKE-be** [comment]
@robodo delegate=IT-Ideas

**Comment by nboulif** [comment]
@IT-Ideas

File: addons/website_sale/static/src/snippets/s_dynamic_snippet_products/000.js [hunk 1]
```diff
@@ -13,25 +13,13 @@ const DynamicSnippetProducts = DynamicSnippetCarousel.extend({
...
-        }).then(nb => nb > 0);
-    },
     /**
      * Method to be overridden in child components if additional configuration elements
      * are required in order to fetch data.
      * @override
      * @private
      */
-    _isConfigComplete: function () {
+    _isConfigComplete: function() {
```
Comment (IT-Ideas) [inline addons/website_sale/static/src/snippets/s_dynamic_snippet_products/000.js:22]: I think this one can stay as function () is correct 🙂

File: addons/website/models/website_snippet_filter.py [hunk 2]
```diff
@@ -70,20 +73,27 @@ def _prepare_values(self, limit=None, search_domain=None):
...
-            return self._filter_records_to_dict_values(records)
+            try:
+                records = self.env[filter_sudo.model_id].search(
+                    domain,
+                    order=','.join(literal_eval(filter_sudo.sort)) or None,
+                    limit=limit
+                )
+                return self._filter_records_to_dict_values(records)
+            except MissingError:
+                _logger.warning("Unexpected domain [%s] in ir.filter, interpreted as False", expression(domain, self).result)
```
Comment (IT-Ideas) [inline addons/website/models/website_snippet_filter.py:84]: Can't we tune the message and just warn that the provided domain caused a MissingError?
@JKE-be, what do you think?

**Comment by IT-Ideas** [comment]
> @IT-Ideas

Thanks for your work 🙂

**Comment by IT-Ideas** [comment]
Thanks @nboulif for your work 🙂.
Just small comments.
I'll test a bit tomorrow 😉

File: addons/website_sale/static/src/snippets/s_dynamic_snippet_products/000.js (hunk 1 above)
Comment (nboulif) [reply addons/website_sale/static/src/snippets/s_dynamic_snippet_products/000.js]: Done :+1:

File: addons/website_sale/static/src/snippets/s_dynamic_snippet_products/000.js [hunk 3]
```diff
@@ -13,18 +13,6 @@ const DynamicSnippetProducts = DynamicSnippetCarousel.extend({
     // Private
     //--------------------------------------------------------------------------
 
-    /**
-     * Check product category exist.
-     * @private
-     * @returns {Promise}
-     */
-    _checkCategoryExist: async function (productCategoryId) {
```
Comment (IT-Ideas) [inline addons/website_sale/static/src/snippets/s_dynamic_snippet_products/000.js:21]: Hello @nboulif 🙂 ,
We should (may) not remove a function or modify a function signature in stable as custom will break.
But as this as been done recently we can take the bet that no-one has made one. Same for removing async etc.

**Comment by IT-Ideas** [comment]
@nboulif I would also squash the two commits 😉

File: addons/website/models/website_snippet_filter.py (hunk 2 above)
Comment (IT-Ideas) [reply addons/website/models/website_snippet_filter.py:84]: _logger.warning("The provided domain [%s] in ir.filter generated a MissingError in '%s'", expression(domain, self).result, self._name)

File: addons/website/models/website_snippet_filter.py [hunk 4]
```diff
@@ -70,20 +73,27 @@ def _prepare_values(self, limit=None, search_domain=None):
...
-            ).sudo().run()
+            try:
+                return self.action_server_id.with_context(
+                    dynamic_filter=self,
+                    limit=limit,
+                    search_domain=search_domain,
+                    get_rendering_data_structure=self._get_rendering_data_structure,
+                ).sudo().run()
+            except MissingError:
+                _logger.warning("Unexpected domain [%s] in action server, interpreted as False", expression(domain, self).result)
```
Comment (IT-Ideas) [inline addons/website/models/website_snippet_filter.py:95]: _logger.warning("The provided domain [%s] in ir.actions.server generated a MissingError in '%s'", expression(domain, self).result, self._name)

File: addons/website/models/website_snippet_filter.py (hunk 2 above)
Comment (nboulif) [reply addons/website/models/website_snippet_filter.py]: Done :+1:

File: addons/website_sale/static/src/snippets/s_dynamic_snippet_products/000.js (hunk 3 above)
Comment (nboulif) [reply addons/website_sale/static/src/snippets/s_dynamic_snippet_products/000.js:21]: Indeed, thanks for the reminder :wink:

File: addons/website/models/website_snippet_filter.py (hunk 4 above)
Comment (nboulif) [reply addons/website/models/website_snippet_filter.py]: Done :+1:

**Comment by nboulif** [comment]
@IT-Ideas Big thanks for the help and review! :wink:

**Comment by IT-Ideas** [comment]
@robodoo r+

**Comment by IT-Ideas** [comment]
@JKE-be , I can't do that 😉

**Comment by JKE-be** [comment]
@robodoo delegate=@IT-Ideas

**Comment by IT-Ideas** [comment]
@robodoo r+

**Pull request closed without merging**